flask/app.py

```diff
@@ -1370,6 +1390,33 @@ def url_defaults(self, f):
         self.url_default_functions.setdefault(None, []).append(f)
         return f
 
+    def _find_error_handler(self, e):
+        """Finds a registered error handler for the request’s blueprint.
+        If neither blueprint nor App has a suitable handler registered, returns None
+        """
+        exc_class, code = self._get_exc_class_and_code(type(e))
+        
+        def find_superclass(d):
+            if not d:
+                return None
+            for superclass in exc_class.mro():
+                if superclass is BaseException:
+                    return None
+                handler = d.get(superclass)
+                if handler is not None:
+                    d[exc_class] = handler  # cache for next time exc_class is raised
```

Comment:
Looks like you have a consensus, so I'm not trying to reopen this, just explaining my arguments:

- When Flask user encounters issues with error handling he is likely to check the contents of `error_handler_spec` during debugging, and be surprised that there are entries there which haven't been explicitly registered.
- On first sight `error_handler_spec` seems to be modified only when a new handler is registered. It's not obvious that we pass one of the inner dicts by reference and modify it inside `_find_error_handler`, especially since the name doesn't suggest that the function has side effects.

Anyway, I think optimizations are worth introducing when there is a real performance problem to solve, and I don't see any problem here... Just my 2 cents.